packages/combo-box/test/keyboard.test.js

```diff
@@ -184,220 +184,273 @@ describe('keyboard', () => {
+        arrowDownKeyDown(input);
 
-      expect(input.value).to.eql('bar');
-    });
+        escKeyDown(input);
 
-    it('should keep selected item on escape when custom value allowed', () => {
-      comboBox.allowCustomValue = true;
-      escKeyDown(input);
-      escKeyDown(input);
-      expect(comboBox.selectedItem).to.eql('bar');
-    });
+        expect(input.value).to.eql('bar');
+      });
 
-    it('should remove selection from the input value selecting value', async () => {
-      arrowDownKeyDown(input);
-      await aTimeout(1);
-      enterKeyDown(input);
+      it('should keep selected item on escape when custom value allowed', () => {
+        comboBox.allowCustomValue = true;
+        escKeyDown(input);
+        escKeyDown(input);
+        expect(comboBox.selectedItem).to.eql('bar');
+      });
 
-      expect(input.selectionStart).to.eql(3);
-      expect(input.selectionEnd).to.eql(3);
-    });
+      it('should remove selection from the input value selecting value', async () => {
+        arrowDownKeyDown(input);
+        await aTimeout(1);
+        enterKeyDown(input);
 
-    it('should not clear the value on esc if clear button is not visible', () => {
-      escKeyDown(input);
-      expect(comboBox.value).to.equal('bar');
-    });
+        expect(input.selectionStart).to.eql(3);
+        expect(input.selectionEnd).to.eql(3);
+      });
 
-    it('should clear the value on esc if clear button is visible', () => {
-      comboBox.close();
-      comboBox.clearButtonVisible = true;
-      escKeyDown(input);
-      expect(comboBox.value).to.equal('');
+      it('should not clear the value on esc if clear button is not visible', () => {
+        escKeyDown(input);
+        expect(comboBox.value).to.equal('bar');
+      });
+
+      it('should clear the value on esc if clear button is visible', () => {
+        comboBox.close();
+        comboBox.clearButtonVisible = true;
+        escKeyDown(input);
+        expect(comboBox.value).to.equal('');
+      });
+
+      it('should not clear the value on esc if the overlay is open', () => {
+        comboBox.clearButtonVisible = true;
+        comboBox.opened = true;
+        escKeyDown(input);
+        expect(comboBox.value).to.equal('bar');
+      });
     });
 
-    it('should not clear the value on esc if the overlay is open', () => {
-      comboBox.clearButtonVisible = true;
-      comboBox.opened = true;
-      escKeyDown(input);
-      expect(comboBox.value).to.equal('bar');
+    describe('auto-open disabled', () => {
+      beforeEach(async () => {
+        comboBox.autoOpenDisabled = true;
+        comboBox.focus();
+        await aTimeout(1);
+      });
+
+      it('should stop propagation of the keyboard enter event when input value is invalid', () => {
+        filter('foobar');
+        const keydownSpy = sinon.spy();
+        document.addEventListener('keydown', keydownSpy);
+        enterKeyDown(input);
+        expect(keydownSpy.called).to.be.false;
```

Comment:
This doesn't check that the `keydown` event is default prevented. Overall, it is the case for pretty much all the new tests.